src/Land/Model/soil_heat.jl

```diff
@@ -1,35 +1,250 @@
+ nodal_update, etc. whenever the prescribed temperature content variables are
+needed by the water model.
 """
-get_temperature(m::PrescribedTemperatureModel) = m.T
+function PrescribedTemperatureModel(T::Function = (aux, t) -> eltype(aux)(0.0))
+    return PrescribedTemperatureModel{typeof(T)}(T)
+end
 
+"""
+    SoilHeatModel{FT, FiT, BCD, BCN} <: AbstractHeatModel
 
+The necessary components for the Heat Equation in a soil water matrix.
 
+# Fields
+$(DocStringExtensions.FIELDS)
 """
-    SoilHeatModel <: AbstractHeatModel
+struct SoilHeatModel{FT, FiT, BCD, BCN} <: AbstractHeatModel
+    "Initial conditions for temperature"
+    initialT::FiT
+    "Dirichlet BC structure"
+    dirichlet_bc::BCD
+    "Neumann BC structure"
+    neumann_bc::BCN
+end
 
-The balance law for internal energy in soil.
+"""
+    SoilHeatModel(
+        ::Type{FT};
+        initialT::FT = FT(NaN),
+        dirichlet_bc::AbstractBoundaryFunctions = nothing,
+        neumann_bc::AbstractBoundaryFunctions = nothing
+    ) where {FT}
+
+Constructor for the SoilHeatModel.
+"""
+function SoilHeatModel(
+    ::Type{FT};
+    initialT = (aux) -> FT(NaN),
+    dirichlet_bc::AbstractBoundaryFunctions = nothing,
+    neumann_bc::AbstractBoundaryFunctions = nothing,
+) where {FT}
+    args = (initialT, dirichlet_bc, neumann_bc)
+    return SoilHeatModel{FT, typeof.(args)...}(args...)
+end
+
+
+#Need these get_temperature functions for SoilHeatModel.
+"""
+    function get_temperature(
+        heat::PrescribedTemperatureModel,
+        aux::Vars,
+        state::Vars,
+        t::Real
+    )
+
+Returns the temperature when the heat model chosen is a user prescribed one.
+
+This is useful for driving Richard's equation without a back reaction on temperature.
+"""
+function get_temperature(
+    heat::PrescribedTemperatureModel,
+    aux::Vars,
+    state::Vars,
+    t::Real,
+)
+    T = heat.T(aux, t)
+    return T
+end
 
-To be used when the user wants to dynamically determine internal energy and temperature.
 
 """
-struct SoilHeatModel <: AbstractHeatModel end
+    function get_initial_temperature(
+        m::PrescribedTemperatureModel,
+        aux::Vars,
+        t::Real
+    )    
+
+Returns the temperature from the prescribed model.
+Needed for soil_init_aux! of SoilWaterModel.
+"""
+function get_initial_temperature(
+    m::PrescribedTemperatureModel,
+    aux::Vars,
+    t::Real,
+)
+    return m.T(aux, t)
+end
+
+
+function get_clima_params_for_heat(land::LandModel, FT)
+    _T_ref = FT(T_0(land.param_set))
+    _LH_f0 = FT(LH_f0(land.param_set))
+
+    _ρ_i = FT(ρ_cloud_ice(land.param_set))
+    _cp_i = FT(cp_i(land.param_set) * _ρ_i)
+
+    _ρ_l = FT(ρ_cloud_liq(land.param_set))
+    _cp_l = FT(cp_l(land.param_set) * _ρ_l)
+
+    return _T_ref, _LH_f0, _ρ_i, _ρ_l, _cp_i, _cp_l
+end
+
+#Probably we dont need κ in aux, unless it is helpful for debugging. 
+
+vars_state(heat::SoilHeatModel, st::Prognostic, FT) = @vars(I::FT)
```

Comment:
What is `I` here? should we use a more verbose name?

Reply:
Also, this could clash with `LinearAlgebra`'s identity matrix `I`, so I'd suggest avoiding this name.

Reply:
we're going to go with U :)!